src/packet.rs

```diff
@@ -53,34 +67,22 @@ impl<W: Write> PacketWriter<W> {
     pub fn end_packet(&mut self) -> io::Result<()> {
         self.maybe_end_packet()
     }
-}
 
-impl<W> PacketWriter<W> {
-    pub fn set_seq(&mut self, seq: u8) {
-        self.seq = seq;
-    }
-}
+    pub fn switch_to_tls(&mut self, config: &TlsConfig) -> io::Result<()> {
+        assert!(self.remaining() == 0); // otherwise we've read ahead into the TLS handshake and will be in trouble.
```

Comment:
Can this actually happen "by accident" in production? If so, we should handle this case more gracefully.

Reply:
If this happens, it would represent a bug. I don't expect it ever could happen, and perhaps can be removed. I will remove it if you prefer.

Reply:
Fair enough — let's leave it as an assertion for now then!